### PR TITLE
added heap sort 

### LINKED ISSUE #87: add heap sort
add heap sort technique

### PR DESCRIPTION
**Pull request by ananyag309**
added heap sort
Fixes #87 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/algorithms/Sorting Algorithms/HeapSort.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+- **Non-Stable**:
+  - Heap sort is a non-stable sorting algorithm, meaning that the relative order of equal elements may not be preserved during the sorting process.
+
+- **No Recursive Calls**:
+  - Unlike recursive algorithms like merge sort, heap sort uses an iterative approach to build the heap and extract elements.
+
+### Time Complexity:
+
+- **Best Case: O(n log n)**  
+  Heap sort involves building the heap (O(n)) and extracting elements from it (O(n log n)), so even in the best case, it requires O(n log n) time.
```
Comment (ajay-dhangar) [inline docs/algorithms/Sorting Algorithms/HeapSort.md]: replace `O(n)` with `$O(n)$` and `O(n log n)` with `$O(n log n)$`

File: docs/algorithms/Sorting Algorithms/HeapSort.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+---
+
+id: selection-sort-algo  
+sidebar_position: 6  
+title: Heap Sort  
+sidebar_label: Heap Sort  
+
```
Comment (ajay-dhangar) [inline docs/algorithms/Sorting Algorithms/HeapSort.md]: add `tags`, and `description` for SEO

File: docs/algorithms/Sorting Algorithms/HeapSort.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+- **No Recursive Calls**:
+  - Unlike recursive algorithms like merge sort, heap sort uses an iterative approach to build the heap and extract elements.
+
+### Time Complexity:
+
+- **Best Case: O(n log n)**  
+  Heap sort involves building the heap (O(n)) and extracting elements from it (O(n log n)), so even in the best case, it requires O(n log n) time.
+
+- **Average Case: O(n log n)**
```
Comment (ajay-dhangar) [inline docs/algorithms/Sorting Algorithms/HeapSort.md:36]: replace `O(n log n)` with `$O(n log n)$`

File: docs/algorithms/Sorting Algorithms/HeapSort.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+### Time Complexity:
+
+- **Best Case: O(n log n)**  
+  Heap sort involves building the heap (O(n)) and extracting elements from it (O(n log n)), so even in the best case, it requires O(n log n) time.
+
+- **Average Case: O(n log n)**  
+  Heap sort's time complexity remains O(n log n) across average cases since the heap operations are logarithmic in nature.
+
+- **Worst Case: O(n log n)**
```
Comment (ajay-dhangar) [inline docs/algorithms/Sorting Algorithms/HeapSort.md:39]: replace `O(n log n)` with `$O(n log n)$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/algorithms/Sorting Algorithms/HeapSort.md (hunk 1 above)
Comment (ananyag309) [reply docs/algorithms/Sorting Algorithms/HeapSort.md]: ok

**Comment by ananyag309** [comment]
@ajay-dhangar is it fine now?

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**